File: 38.cpp
class Solution {
public:
    string countAndSay(int n) {
        if (n == 0) {
            return "";
        }
        string s = "";
        for(int i = 0; i < n; i++) {
            s = countAndSay(s);
        }
        return s;
    }
    string countAndSay(string& s) {
        string output;
        if (s.size() == 0){
            return "1";
        }
        int count = 0;
        char prev = s[0];
        for (char c : s) {
            if (prev == c) {
                count ++;
            } else {
                output += char(count + '0');
                output += prev;
                prev = c;
                count = 1;
            }
        }
        output += char(count + '0');
        output += prev;
        return output;
    }
};

File: 38.py
class Solution:
    def countAndSay(self, n: int) -> str:
        def countAndSayStr(s: str) -> str:
            if not s:
                return "1"
            output = []
            prev = s[0]
            count = 0
            for c in s:
                if c == prev:
                    count += 1
                else:
                    output.append(chr(ord('0') + count))
                    output.append(prev)
                    prev = c
                    count = 1
            output.append(chr(ord('0') + count))
            output.append(prev)
            return "".join(output)
        
        if n == 0:
            return ""
        else:
            s = ""
            for _ in range(n):
                s = countAndSayStr(s)
            return s
